java-util/src/main/java/io/druid/java/util/common/granularity/DurationGranularity.java

```diff
@@ -66,20 +68,39 @@ public long getOriginMillis()
   }
 
   @Override
-  public long next(long t)
+  public DateTimeFormatter getFormatter(Formatter type)
   {
-    return t + getDurationMillis();
+    throw new RE("This method should not be invoked for this granularity type");
   }
 
   @Override
-  public long truncate(final long t)
+  public DateTime increment(DateTime time)
   {
+    return new DateTime(time.getMillis() + getDurationMillis());
+  }
+
+  @Override
+  public DateTime decrement(DateTime time)
+  {
+    return new DateTime(time.getMillis() - getDurationMillis());
+  }
+
+  @Override
+  public DateTime truncate(DateTime time)
```

Comment:
also an [existing code](https://github.com/druid-io/druid/blob/master/processing/src/main/java/io/druid/granularity/DurationGranularity.java#L75) and this wasn't changed in this PR